src/pudl/extract/dbf.py

```diff
@@ -181,28 +181,34 @@ def load_table(self, table_name: str) -> pd.DataFrame:
 class AbstractFercDbfReader(Protocol):
     """This is the interface definition for dealing with fox-pro datastores."""
 
-    def get_dataset(self) -> str:
+    def get_dataset(self: Self) -> str:
         """Returns name of the dataset that this datastore provides access to."""
         ...
 
-    def get_table_names(self) -> list[str]:
+    def get_table_names(self: Self) -> list[str]:
         """Returns list of all available table names."""
         ...
 
-    def get_archive(self, **filters) -> FercDbfArchive:
+    def get_archive(self: Self, **filters) -> FercDbfArchive:
         """Returns single archive matching specific filters."""
         ...
 
-    def get_table_schema(self, table_name: str, year: int) -> DbfTableSchema:
+    def get_table_schema(self: Self, table_name: str, year: int) -> DbfTableSchema:
         """Returns schema for a given table and a given year."""
         ...
 
     def load_table_dfs(
-        self, table_name: str, partitions: list[dict[str, Any]]
+        self: Self, table_name: str, partitions: list[dict[str, Any]]
     ) -> pd.DataFrame | None:
         """Returns dataframe that contains data for a given table across given years."""
         ...
 
+    def transform_table_part(
```

Comment:
Hm. This shows the limitation of the current architecture. I hoped that we can contain all dataset-specific logic in the FercDbfExtractor class and then rely on a single `FercDbfReader`. Now, we have `transform_table` in the Extractor but `transform_table_part` in the `FercDbfReader` which feels less than ideal.

I can think of two alternatives from the top of my head:
1. pass `df_transformer: Callable[...]` to `load_table_dfs` method and pass `FercDbfExtractor.transform_table_part` to it. This feels a bit off, because the dependencies go both ways, so the layering is not quite right.
2. move the logic for concatenating dfs out into `FercDbfExtractor`. e.g. instead of `load_table_dfs` returning single df, it could return list of tuples `(partition, df)` and that would be aggregated in extactor.

(2) might be the most straightforward approach for this IMO.

Reply:
I agree (2) sounds better. Would you be willing to take a stab at implementing this? I think I need to focus on the XBRL calculations/metadata stuff.

Reply:
Sure, let me try to take a stab at this.